### PR TITLE
Switch to standard lib ServeMux

### PR DESCRIPTION
**Pull request by sashankg**
Removes unnecessary dependency + makes it easier to implement serving static directory for FE. 

Also refactored the handlers tests to directly call the handler instead of setting up a server to route the requests to better isolate.

### CODE REVIEW

**Review by eagraf — COMMENTED** [review]
Looks great. Should we wait for #10, or merge this first?

**Comment by sashankg** [comment]
> Looks great. Should we wait for #10, or merge this first?
 
Yeah let's wait just to keep things easy

File: internal/node/controller/routes.go [hunk 1]
```diff
@@ -234,11 +217,6 @@ func (h *AddUserRoute) Method() string {
 }
 
 func (h *AddUserRoute) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	if r.Method != http.MethodPost {
```
Comment (arushibandi) [inline internal/node/controller/routes.go:237]: why did these checks get deleted?

**Review by arushibandi — COMMENTED** [review]
I think using `gorilla` is pretty standard and has some advanced routing feature that we might want in the future, but down to merge this for now

File: internal/node/controller/routes.go (hunk 1 above)
Comment (arushibandi) [reply internal/node/controller/routes.go:237]: oh nvm see this got moved to `processRoute`

**Review by arushibandi — APPROVED** [review]

**Pull request merged**